bot/src/data-updater.ts

```diff
@@ -55,6 +62,42 @@ export default class DataUpdater {
 						}
 					]).toArray();
 				}
+			},
+			{
+				toVersion: 2,
+				run: async () => {
+					const users = this.databaseClient.userRepository;
+					await this.legacyDB.connect();
+					const legacyUsers = this.legacyDB.userRepository;
+
+					const allLegacyUsers = await legacyUsers.getAll();
+
+					for (let legacyUser of allLegacyUsers) {
+						const user = await users.get(legacyUser.discordID);
+						if (user.legacyData) continue;
+						user.legacyData = {
+							experience: legacyUser.experience,
+							hostedSessionsDuration: legacyUser.hostedSessionsDuration,
+							joinedSessionsDuration: legacyUser.joinedSessionsDuration,
+							sessionsHosted: legacyUser.sessionsHosted,
+							sessionsJoined: legacyUser.sessionsJoined,
+						}
+						await users.update(user);
```

Comment:
If an error is thrown after this line then the user won't get their recommendation score but they have their legacy data so it will look like they have been updated.
Could wrap each "user update" into a transaction to prevent that from happening.